packages/angular_devkit/schematics/src/tree/recorder.ts

```diff
@@ -50,3 +67,22 @@ export class UpdateRecorderBase implements UpdateRecorder {
     return this._content.generate();
   }
 }
+
+
+export class UpdateRecorderBom extends UpdateRecorderBase {
+  constructor(entry: FileEntry, private _delta = 3) {
+    super(entry);
+  }
+
+  insertLeft(index: number, content: Buffer | string) {
```

Comment:
Is `index` in terms of byte offsets or character offsets?  Also is the string for `content` converted into the encoding specified by the BOM?  It might be useful longer term to have both a `BinaryUpdateRecorder` and `TextUpdateRecorder`.

Reply:
Indeed. Right now it's in term of binary offset. I'm still unsure how we want to approach this problem.

Reply:
We need an `UpdateBuffer` that supports text instead of buffers. That's not a huuuuge challenge, but something we should defer to the libraries (maybe?).